src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -0,0 +1,179 @@
+    }
+
+    if (channel?.type === 'topics') return;
+
+    try {
+      await Promise.all(
+        (channel?.members || []).map(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+            channel?.id,
+            uuid(),
+            member
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+          await memberSchema.save(localDb);
+        })
+      );
+
+      await Promise.all(
+        (channel?.messages || []).map(async (message) => {
+          if (message?.type === 'deleted') return;
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          await chat.save(localDb);
+        })
+      );
+    } catch (e) {
+      console.log('error on saveChannelData:', e);
+    }
+  };
+
+  const saveAllChannelData = async (channels, channelCategory: ChannelCategory) => {
+    const filteredChannels = channels.filter((channel) => {
+      const isLocationChannel = channel?.channel_type === 2 || channel?.type_channel === 2;
+      const isDeletedMessage = channel?.firstMessage?.type === 'deleted';
+      const isSelfChatChannel = channel?.type === 'messaging' && channel?.members?.length < 2;
+      const isCommunityChannel = channel?.type === 'topics';
+      const isDeletedCommunityMessage = channel?.messages[channel?.messages?.length - 1]?.text
+        ?.toLowerCase()
+        ?.includes('this message was deleted');
+
+      return (
+        !isLocationChannel &&
+        !(isDeletedMessage || isSelfChatChannel) &&
+        !((isDeletedMessage || isDeletedCommunityMessage) && isCommunityChannel)
+      );
+    });
+
+    filteredChannels.forEach(async (channel) => {
+      await saveChannelData(channel, channelCategory);
+    });
+  };
+
+  const getAllSignedChannels = async () => {
+    if (!localDb) return;
+    let signedChannel;
+
+    try {
+      signedChannel = await SignedMessageRepo.getAllSignedChannels();
+    } catch (e) {
+      console.log('error on getting signedChannel:', e);
+    }
+
+    try {
+      await saveAllChannelData(signedChannel ?? [], 'SIGNED');
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving signedChannel:', e);
+    }
+  };
+
+  const getAllAnonymousChannels = async () => {
+    if (!localDb) return;
+    let anonymousChannel: ChannelData[] = [];
+    try {
+      anonymousChannel = await AnonymousMessageRepo.getAllAnonymousChannels();
+    } catch (e) {
+      console.log('error on getting anonymousChannel:', e);
+    }
+
+    try {
+      await saveAllChannelData(anonymousChannel, 'ANONYMOUS');
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving anonymousChannel:', e);
+    }
+  };
+
+  return {
+    getAllSignedChannels,
+    getAllAnonymousChannels
+  };
+};
+
+export default useFetchChannelHook;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useFetchChannelHook` function seems to be well implemented. However, there are a few areas that could be improved for better error handling and maintainability:

1. Error Handling: Currently, the code is using `console.log` to log errors. It would be more appropriate to use a proper logging mechanism or an error reporting service which can provide more context about the error, such as stack traces, timestamps, etc.

2. Async/Await: In the `saveAllChannelData` function, you're using `forEach` with async functions. This will not work as expected because `forEach` will not wait for promises to resolve before moving on to the next iteration. Instead, consider using `for...of` loop or `Promise.all`.

3. Code Duplication: The `getAllSignedChannels` and `getAllAnonymousChannels` functions have similar structures. You could create a single function that accepts the necessary differences as parameters to reduce code duplication.

Here are the suggested changes:

```diff
-  filteredChannels.forEach(async (channel) => {
-    await saveChannelData(channel, channelCategory);
-  });
+  for (const channel of filteredChannels) {
+    await saveChannelData(channel, channelCategory);
+  }

-  const getAllSignedChannels = async () => {
-    if (!localDb) return;
-    let signedChannel;
-
-    try {
-      signedChannel = await SignedMessageRepo.getAllSignedChannels();
-    } catch (e) {
-      console.log('error on getting signedChannel:', e);
-    }
-
-    try {
-      await saveAllChannelData(signedChannel ?? [], 'SIGNED');
-      refresh('channelList');
-    } catch (e) {
-      console.log('error on saving signedChannel:', e);
-    }
-  };
-
-  const getAllAnonymousChannels = async () => {
-    if (!localDb) return;
-    let anonymousChannel: ChannelData[] = [];
-    try {
-      anonymousChannel = await AnonymousMessageRepo.getAllAnonymousChannels();
-    } catch (e) {
-      console.log('error on getting anonymousChannel:', e);
-    }
-
-    try {
-      await saveAllChannelData(anonymousChannel, 'ANONYMOUS');
-      refresh('channelList');
-    } catch (e) {
-      console.log('error on saving anonymousChannel:', e);
-    }
-  };

+  const getAllChannels = async (repo, channelCategory) => {
+    if (!localDb) return;
+    let channels;
+
+    try {
+      channels = await repo.getAllChannels();
+    } catch (e) {
+      console.log('error on getting channels:', e);
+    }
+
+    try {
+      await saveAllChannelData(channels ?? [], channelCategory);
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving channels:', e);
+    }
+  };
+
+  return {
-    getAllSignedChannels,
-    getAllAnonymousChannels
+    getAllChannels: (repo, channelCategory) => getAllChannels(repo, channelCategory)
  };
```

Please replace the `console.log` with a proper logging mechanism or error reporting service as per your project's standards.


<!-- This is an auto-generated comment by OSS CodeRabbit -->